File: csharp/src/Core/Models/HTLCModels/HTLCAddLockSigTransactionPrepareRequest.cs
namespace Train.Solver.Core.Models.HTLCModels;

public class HTLCAddLockSigTransactionPrepareRequest
{
    public string Id { get; set; } = null!;

    public string Hashlock { get; set; } = null!;

    public long Timelock { get; set; }

    public string? R { get; set; }

    public string? S { get; set; }

    public string? V { get; set; }

    public string? Signature { get; set; }

    public string Asset { get; set; } = null!;

    public string[]? SignatureArray { get; set; }
}

File: js/src/CoreAbstraction/Models/TransactionBuilderModels/HTLCAddLockSigTransactionPrepareRequest.ts
export interface HTLCAddLockSigTransactionPrepareRequest {
    Id: string;
    Hashlock: string;
    Timelock: number;  
    R?: string;        
    S?: string;        
    V?: string;        
    Signature?: string;
    Asset: string;
    SignatureArray?: string[];
  }